src/plugins/simpleini/simpleini.c

```diff
@@ -152,7 +152,7 @@ static char * replaceStringFormatSpec (char * format, const char * replace)
 		// copy pre replacement
 		strncpy (result, format, preReplLen);
 		// copy replacement
-		strncpy (result + preReplLen, replace, replaceLen);
+		strcpy (result + preReplLen, replace);
```

Comment:
Iirc his causes a warning on OpenBSD.

Reply:
While, this might be the case we do

- not have a build job that uses OpenBSD, and
- the current version of the code causes a warning on a non-outdated version of GCC

. I also do not think using `strcpy` should be a problem, since we already make sure that the buffer filled by the function is large enough.

If you have a better solution to fix the warning, I am also very open to just use that one instead.

Reply:
Thank you for fixing the warning! Unfortunately I have not seen the warning yet, so I do not know what the problem was. What do you think about pasting the warning text into the commit messages?

It is very nice that you have many separate commits.

> not have a build job that uses OpenBSD

Yes, that is the reason why I needed to say it in the review :smile: 

Anyway, sometimes there are situations where it is impossible to fix a warning for every system. So maybe we need to suppress this warning in OpenBSD once somebody takes care of that system.

Reply:
> Unfortunately I have not seen the warning yet, so I do not know what the problem was.

The build system emits the warning, if you [build Elektra on Debian Unstable](https://build.libelektra.org/jenkins/blue/organizations/jenkins/libelektra/detail/master/296/pipeline/420/#step-706-log-379):

```
src/plugins/simpleini/simpleini.c: In function ‘replaceStringFormatSpec’:
src/plugins/simpleini/simpleini.c:155:3: warning: ‘strncpy’ output truncated before terminating nul copying as many bytes from a string as its length [-Wstringop-truncation]
   strncpy (result + preReplLen, replace, replaceLen);
   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/plugins/simpleini/simpleini.c:142:22: note: length computed here
  size_t replaceLen = strlen (replace);
                      ^~~~~~~~~~~~~~~~
```

.

> What do you think about pasting the warning text into the commit messages?

While I am in general in favor of adding information to commit messages, the warning above is quite common. Searching for “string truncation GCC” on the web should return enough information, if you want to read more about the warning.